src/Microsoft.AspNetCore.Http.Connections/HttpConnectionManager.cs

```diff
@@ -114,118 +104,87 @@ private static string MakeNewConnectionId()
             return WebEncoders.Base64UrlEncode(buffer);
         }
 
-        private static void Scan(object state)
+        private async Task ExecuteTimerLoop()
         {
-            ((HttpConnectionManager)state).Scan();
-        }
-
-        public void Scan()
-        {
-            // If we couldn't get the lock it means one of 2 things are true:
-            // - We're about to dispose so we don't care to run the scan callback anyways.
-            // - The previous Scan took long enough that the next scan tried to run in parallel
-            // In either case just do nothing and end the timer callback as soon as possible
-            if (!Monitor.TryEnter(_executionLock))
+            using (_timer)
```

Comment:
I'm not sure what this using statement is doing. Would a comment explaining what is going on be useful?

Reply:
Part of the goal here is to change the timer callback into a loop, so in this case we have a concrete, lexical, lifetime for the timer which means we can use `using`